generator/template.go

```diff
@@ -195,6 +195,91 @@ function getNotifyEntityArrivalSink<T>(notifyCallback: NotifyStreamEntityArrival
     }
   })
 }
+
+/**
+ * Flattens a deeply nested object
+ *
+ * @param  {Object} deeplyNestedObject
+ * @param  {String} path
+ * @return {Record<string, unknown>}
+ */
+function flattenObject<T extends Record<string, unknown>>(
+  deeplyNestedObject: T,
+  path: string = ""
+): T {
+  return Object.keys(deeplyNestedObject).reduce(
+    (acc: T, key: string): T => {
+      const value = deeplyNestedObject[key];
+
+      const newPath = Array.isArray(deeplyNestedObject)
+        ? path + "." + key
+        : [path, key].filter(Boolean).join(".");
+      const isObject = [
+        typeof value === "object",
+        value !== null,
+        !(value instanceof Date),
+        !(value instanceof RegExp),
+        !(Array.isArray(value) && value.length === 0)
+      ].every(Boolean);
+
+      return isObject
+        ? { ...acc, ...flattenObject(value as Record<string, unknown>, newPath) }
+        : { ...acc, [newPath]: value };
+    },
+    {} as T
+  );
+}
+
+/**
+ * Renders a deeply nested object into a string of URL search parameters
+ * by first flattening the object and then removing non-primitive array keys,
+ * non-primitive values and keys which are already present in the URL path.
+ * @param  {Object} deeplyNestedObject
+ * @param  {Array<string>} urlPathParams
+ * @return {String}
+ */
+export function renderURLSearchParams<T extends Record<string, unknown>>(
+  deeplyNestedObject: T,
+  urlPathParams: string[] = []
+): string {
+  const flattenedObject = flattenObject(deeplyNestedObject);
+
+  const urlSearchParams = Object.keys(flattenedObject).reduce(
```

Comment:
I have used `reduce` here because renderURLSearchParams takes the whole request object as its input, which contains the key that may be present in the URL path as a parameter. So we need to ensure that fields in the request object that are present as URL path parameters are removed.

Reply:
once the complex logic is removed. I guess you could just chain the calls like 

`Object.keys(flattenedObject).filter(keysInPath).map()`

Reply:
Aaah! Indeed. Thank you for your suggestion. I will try this out 👍🏼